### PR TITLE
Upgrade Elixir to 1.15.5 And Erlang to 24.3.4

### PR DESCRIPTION
**Pull request by Kwilliams-dev303**
## Description
Updating elixir, erlang, phoenix and relative dependencies to more current versions. Only new dependency is `{:phoenix_view, "~> 2.0"},` and it resolved the following error. 

```
error: module Phoenix.View is not loaded and could not be found
  lib/push_ex_web/views/error_view.ex:4: PushExWeb.ErrorView (module)
```

## Motivation and Context
This change is required as part of a elixir upgrade initiative for repos that use this dependency.

## How Has This Been Tested?
Core logic is unchanged, so testing consisted of running the current tests on repeat. A couple of flaky tests were spotted but the same behavior was shown on master. 

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Elixir upgrade

## Checklist:
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by sb8244** [comment]
I like this. I will take a look to make sure nothing breaks. Part of the base of this library was dealing with quirks that were identified in the original versions in use. With an upgrade, those quirks may have changed and potentially broken something. Especially true on the JS side.

**Comment by Kwilliams-dev303** [comment]
> I like this. I will take a look to make sure nothing breaks. Part of the base of this library was dealing with quirks that were identified in the original versions in use. With an upgrade, those quirks may have changed and potentially broken something. Especially true on the JS side.

Gotcha. Thanks for taking a look!

**Comment by Kwilliams-dev303** [comment]
@sb8244 I see that a lot of the builds have failed for this PR. Is this expected?

**Comment by sb8244** [comment]
It's not expected @Kwilliams-dev303 . The issue is that this removes support for old OTP and Elixir versions, which means those won't pass the test matrix.

Are you able to update the matrix to reflect proper versions? https://github.com/pushex-project/pushex/blob/master/.github/workflows/elixir.yml#L15

There may also be issues where I need to update the workflow to remove deprecated build pipeline

**Comment by sb8244** [comment]
I strongly believe that Ubuntu 18 will not work. Here is one of my current workflows that does work. We would need to modify it to support the text matrix and same testing regimen as this repo has:

```yaml
name: Elixir CI

on:
  pull_request:

jobs:
  build:
    name: Build and test
    runs-on: ubuntu-22.04
    env:
      MIX_ENV: test
      ELIXIR_ENV: test
      DATABASE_USER: postgres
      DATABASE_PASSWORD: postgres
      DATABASE_PORT: 5432
      DATABASE_HOST: localhost
    services:
      postgres:
        image: postgres:10.10
        env:
          POSTGRES_PASSWORD: "postgres"
        ports:
          - 5432:5432
        options: >-
          --health-cmd pg_isready
          --health-interval 10s
          --health-timeout 5s
          --health-retries 5

    steps:
      - uses: actions/checkout@v2
      - name: Set up Elixir
        uses: erlef/setup-beam@v1
        with:
          elixir-version: "1.15.7" # Define the elixir version [required]
          otp-version: "26.1.2" # Define the OTP version [required]

      - name: Restore dependencies cache
        uses: actions/cache@v2
        with:
          path: deps
          key: ${{ runner.os }}-mix-${{ hashFiles('**/mix.lock') }}
          restore-keys: ${{ runner.os }}-mix-

      - name: Cache _build
        uses: actions/cache@v1
        id: build-cache
        with:
          path: _build
          key: ${{ runner.os }}-build-${{ matrix.otp }}-${{ matrix.elixir }}-${{ hashFiles(format('{0}{1}', github.workspace, '/mix.lock')) }}

      - name: Install dependencies
        run: mix deps.setup && mix deps.get

      - name: Compile
        run: MIX_ENV=test mix compile --warnings-as-errors

      - name: Run tests
        run: mix test --exclude feature

      - name: Formatter
        run: mix format --check-formatted

      - name: Install asset deps
        run: mix assets.install

      - name: Prettier check
        run: mix assets.check_format

      - name: Sobelow check
        run: mix sobelow --exit --skip
```

**Comment by Kwilliams-dev303** [comment]
> It's not expected @Kwilliams-dev303 . The issue is that this removes support for old OTP and Elixir versions, which means those won't pass the test matrix.
> 
> Are you able to update the matrix to reflect proper versions? https://github.com/pushex-project/pushex/blob/master/.github/workflows/elixir.yml#L15
> 
> There may also be issues where I need to update the workflow to remove deprecated build pipeline

Thanks for the quick reply @sb8244. Looking into updating the most recent versions in the matrix now 👀

**Comment by Kwilliams-dev303** [comment]
@sb8244 Let me know if my most recent commit was along the lines of what you were thinking ☝️

File: .github/workflows/elixir.yml [hunk 1]
```diff
@@ -6,40 +6,39 @@ on:
...
-    runs-on: ubuntu-18.04
+    runs-on: ubuntu-22.04
     env:
       MIX_ENV: test
       ELIXIR_ENV: test
     strategy:
       matrix:
-        otp: ['22.3.4', '23.2.7', '24.1']
-        elixir: ['1.11.4', '1.12.3']
+        otp: ["22.3.4", "23.2.7", "24.1", "26.1.2"]
```
Comment (sb8244) [inline .github/workflows/elixir.yml:15]: @Kwilliams-dev303 can you update this to only include:

```
        otp: ["24.1", "26.1.2"]
        elixir: ["1.15.7"]
```

This release would drop support for all lower Elixir versions than 1.15.7

Comment (Kwilliams-dev303) [reply .github/workflows/elixir.yml:15]: @sb8244 sure thing

Comment (sb8244) [reply .github/workflows/elixir.yml:15]: And otp should actually be `24.3.4` and not `24.1`. The new builders require specific patch numbers

Comment (Kwilliams-dev303) [reply .github/workflows/elixir.yml]: @sb8244 gotcha 👍 pushing change

Comment (sb8244) [reply .github/workflows/elixir.yml]: Booms tests are running now!! Fingers crossed 🤞

Comment (Kwilliams-dev303) [reply .github/workflows/elixir.yml]: Ayy! Looks like we got one failure though. It passes locally for me.

**Comment by sb8244** [comment]
I'm going to merge this into upgrade-elixir so I can make any changes as needed. Looks like flake test so not a problem there.

**Pull request merged**